apps/root/src/pages/transfer/transfer-container/components/recipient-address/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { useAppDispatch } from '@hooks/state';
+import useActiveWallet from '@hooks/useActiveWallet';
+import useReplaceHistory from '@hooks/useReplaceHistory';
+import useCurrentNetwork from '@hooks/useCurrentNetwork';
+import { setRecipient } from '@state/transfer/actions';
+import { useTransferState } from '@state/transfer/hooks';
+import { defineMessage, useIntl } from 'react-intl';
+import { DescriptionOutlinedIcon, IconButton, InputAdornment, TextField } from 'ui-library';
+import { validateAddress } from '@common/utils/parsing';
+import ContactListModal from './components/contact-list-modal';
+
+interface RecipientAddressProps {
+  recipientParam?: string;
+}
+
+const RecipientAddress = ({ recipientParam }: RecipientAddressProps) => {
+  const intl = useIntl();
+  const dispatch = useAppDispatch();
+  const replaceHistory = useReplaceHistory();
+  const activeWallet = useActiveWallet();
+  const [recipientInput, setRecipientInput] = React.useState<string>('');
+  const { token, recipient: storedRecipient } = useTransferState();
+  const currentNetwork = useCurrentNetwork();
+  const [shouldShowContactList, setShouldShowContactList] = React.useState<boolean>(false);
+
+  const inputRegex = RegExp(/^[A-Fa-f0-9x]*$/);
+  const { isValidRecipient } = validateAddress(recipientInput, activeWallet?.address);
+
+  React.useEffect(() => {
+    const { isValidRecipient: isValid } = validateAddress(recipientParam || '', activeWallet?.address);
+    if (recipientParam && isValid) {
+      setRecipientInput(recipientParam);
+      dispatch(setRecipient(recipientParam));
+    }
+  }, [dispatch, validateAddress]);
```

Comment:
no need to pass validateAddress, but it should change based on activeWallet.address and recipientParam

Reply:
Or is this for when we load it the first time?

Reply:
Yes, this hook is triggered during the initial load. I've made some changes, though, and now everything is consolidated within the `TransferForm`.